OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -0,0 +1,241 @@
+
+		public override void Tick()
+		{
+			var actor = editor.DefaultBrush.SelectedActor;
+			if (actor != null)
+			{
+				var origin = worldRenderer.Viewport.WorldToViewPx(new int2(actor.Bounds.X, actor.Bounds.Y));
+
+				// If we scrolled, hide the edit box for a moment
+				if (actorSelectBorder.Bounds.X != origin.X || actorSelectBorder.Bounds.Y != origin.Y)
+				{
+					lastScrollTime = Game.RunTime;
+				}
+
+				// If we changed actor, move widgets
+				if (currentActor != actor)
+				{
+					lastScrollTime = 0; // Ensure visible
+					selectedOwner = editorActorLayer.Players.Players.Values.First(x => x.Name == actor.Owner.Name);
+					ownersDropDown.Text = selectedOwner.Name;
+					ownersDropDown.TextColor = selectedOwner.Color.RGB;
+
+					if (currentActor != null) // Hide border of the previously selected actor.
+						currentActor.ShowBorder = false;
+					currentActor = actor;
+					currentActor.ShowBorder = true;
+
+					actorIdField.Text = actor.ID;
+					typeLabel.Text = actor.TypeSummary;
+					actorIdField.CursorPosition = actor.ID.Length;
+					actorSelectBorder.Bounds.Width = actor.Bounds.Width * 2;
+					actorSelectBorder.Bounds.Height = actor.Bounds.Height * 2;
+					HideValidationMessage();
+				}
+
+				actorSelectBorder.Bounds.X = origin.X;
+				actorSelectBorder.Bounds.Y = origin.Y;
+
+				// Set the edit panel to the right of the selection border.
+				int newX = origin.X + actorSelectBorder.Bounds.Width + editPanelPadding;
+				actorEditPanel.Bounds.X = newX;
+				actorEditPanel.Bounds.Y = origin.Y;
+			}
+			else
+			{
+				// Selected actor is null, hide the border and edit panel.
+				if (currentActor != null)
+					currentActor.ShowBorder = false;
+
+				actorIdField.YieldKeyboardFocus();
+				currentActor = null;
+			}
+		}
+
+		void ShowValidationMessage(string message)
+		{
+			actorEditPanel.Bounds.Height = validationHeight;
+			validationLabel.Text = message;
+			validationLabel.Visible = true;
+			actorIdField.TakeKeyboardFocus();
+		}
+
+		void HideValidationMessage()
+		{
+			actorEditPanel.Bounds.Height = originalHeight;
+			validationLabel.Visible = false;
+		}
+
+		void Save()
+		{
+			if (currentActor == null)
+				return;
+
+			var newActorName = actorIdField.Text;
+			var newActorNameLower = newActorName.ToLowerInvariant();
+
+			if (string.IsNullOrWhiteSpace(newActorName))
+			{
+				ShowValidationMessage("Please enter an actor ID.");
+				return;
+			}
+
+			// Check for duplicate actor ID
+			if (currentActor.ID.ToLowerInvariant() != newActorNameLower)
+			{
+				bool found = world.Map.ActorDefinitions.Any(x => x.Key.ToLowerInvariant() == newActorNameLower);
+				if (found)
+				{
+					ShowValidationMessage("An actor is already using this ID.");
+					return;
+				}
+			}
+
+			// Change key in actor defs
+			var actorDef = world.Map.ActorDefinitions.First(x => x.Key == currentActor.ID);
+			actorDef.Key = newActorName;
+
+			// Remove existing editor actor
+			editorActorLayer.Remove(currentActor);
```

Comment:
You should be able to `currentActor.ReplaceInit(new OwnerInit(selectedOwner.Name));` instead of serializing/removing/editing/readding the actor.